conda_libmamba_solver/shards_subset.py

```diff
@@ -234,11 +253,146 @@ def shortest_bfs(self, root_packages):
                     if not next_node.visited:
                         node_queue.append(next_node)
 
+    def shortest_pipelined(self, root_packages):
+        """
+        Build repodata subset using a main thread, a thread to fetch from
+        sqlite3 and another threadpool to fetch http.
+        """
+
+        # these are in self.nodes but we still need the shards
+        self.nodes = dict(_nodes_from_packages(root_packages, self.shardlikes))
+
+        # if sharded is empty, we could skip everything related to threads.
+        sharded = [s for s in self.shardlikes if isinstance(s, Shards)]
+        cache = sharded[0].shards_cache if sharded else None
+
+        cache_in_queue: SimpleQueue[list[NodeId] | None] = SimpleQueue()
+        shard_out_queue: SimpleQueue[list[tuple[NodeId, ShardDict]]] = SimpleQueue()
+        cache_miss_queue: SimpleQueue[list[NodeId] | None] = SimpleQueue()
+
+        # this kind of thread can crash, and we don't hear back without our own
+        # handling.
+        cache_thread = threading.Thread(
+            target=cache_fetch_thread,
+            args=(cache_in_queue, shard_out_queue, cache_miss_queue, cache),
+            daemon=False,
+        )
+
+        # this kind of thread can crash, and we don't hear back without our own
+        # handling.
+        network_thread = threading.Thread(
+            target=network_fetch_thread,
+            args=(cache_miss_queue, shard_out_queue, cache, self.shardlikes),
+            daemon=False,
+        )
+
+        # or after populating initial URLs queue
+        cache_thread.start()
+        network_thread.start()
+
+        # see test_*_thread in test_shards.py for how to set up the workers
+
+        shardlikes_by_url = {s.url: s for s in self.shardlikes}
+        pending = set(self.nodes)  # to be sent to in_queue
+        submitted = set()  # sent to in_queue, not received from shard_out_queue
+        processed = set()  # received from shard_out_queue
+
+        def enqueue_pending():
+            log.info(
```

Comment:
```suggestion
            log.debug(
```
Something seems to be removing log handlers even though I call `basicConfig()` in a fixture. Maybe a different version of pytest or a plugin? `pytest --log-cli-level=DEBUG` works.